src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
```

Comment:
PrimathonAICodeReviewer

Consider adding error handling in the intercept method to catch and log any errors that occur during the interception process.

```diff
+     intercept(
+       context: ExecutionContext,
+       next: CallHandler,
+     ): Observable<Response<T>> {
+       try {
```



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->